From daaf9818786f22649890fc1400c04a87dc987329 Mon Sep 17 00:00:00 2001
From: "alex.cao" <alex.cao@droid01-sz.amlogic.com>
Date: Tue, 27 May 2014 18:01:50 +0800
Subject: [PATCH 4301/5965] PD #87697 add P2P play for MX 1,Increase the osd0
 framebuffer size to 1280*1080 2,The font have sawtooth when disable freescale
 in 480i/576i

---
 arch/arm/boot/dts/amlogic/meson6_g18.dtd      | 4 ++--
 arch/arm/boot/dts/amlogic/meson6_g18_512M.dtd | 4 ++--
 drivers/amlogic/display/osd/osd_hw.c          | 2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/meson6_g18.dtd b/arch/arm/boot/dts/amlogic/meson6_g18.dtd
index 5560a4038f29..d715d7418dbc 100755
--- a/arch/arm/boot/dts/amlogic/meson6_g18.dtd
+++ b/arch/arm/boot/dts/amlogic/meson6_g18.dtd
@@ -111,11 +111,11 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-		reserve-memory = <0x00b00000 0x00100000>;
+		reserve-memory = <0x01000000 0x00100000>;
 		reserve-iomap = "true";
 	  vmode = <0>; /*0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 	  scale_mode = <0>; /*0:default 1:new*/
-	  display_size_default = <1280 720 1280 2160 32>;  // osd0:8M, osd1:1m  1280*720*4*3 = 11,059,200
+	  display_size_default = <1280 1080 1280 3240 32>;  // osd0:16M, osd1:1m  1280*1080*4*3 = 16,588,800
 	};
 //$$ DEVICE="deinterlace"
 //$$ L2 PROP_STR = "status"
diff --git a/arch/arm/boot/dts/amlogic/meson6_g18_512M.dtd b/arch/arm/boot/dts/amlogic/meson6_g18_512M.dtd
index ea22fd34fdbd..5b35c8c589d2 100755
--- a/arch/arm/boot/dts/amlogic/meson6_g18_512M.dtd
+++ b/arch/arm/boot/dts/amlogic/meson6_g18_512M.dtd
@@ -111,11 +111,11 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-		reserve-memory = <0x00b00000 0x00100000>;
+		reserve-memory = <0x01000000 0x00100000>;
 		reserve-iomap = "true";
 	  vmode = <0>; /*0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 	  scale_mode = <0>; /*0:default 1:new*/
-	  display_size_default = <1280 720 1280 2160 32>;  // osd0:8M, osd1:1m  1280*720*4*3 = 11,059,200
+	  display_size_default = <1280 1080 1280 3240 32>;  // osd0:16M, osd1:1m  1280*1080*4*3 = 16,588,800
 	};
 //$$ DEVICE="deinterlace"
 //$$ L2 PROP_STR = "status"
diff --git a/drivers/amlogic/display/osd/osd_hw.c b/drivers/amlogic/display/osd/osd_hw.c
index 0969a0677f1c..a97147c79e56 100755
--- a/drivers/amlogic/display/osd/osd_hw.c
+++ b/drivers/amlogic/display/osd/osd_hw.c
@@ -291,7 +291,7 @@ static irqreturn_t osd_rdma_isr(int irq, void *dev_id)
 				}
 			}
 		} else {
-			if ((osd_hw.pandata[OSD1].y_start%2) == 0){
+			if ((osd_hw.pandata[OSD1].y_start%2) == 1){
 				odd_or_even_line = (aml_read_reg32(P_ENCI_INFO_READ) & (1<<29)) ?
 									OSD_TYPE_BOT_FIELD : OSD_TYPE_TOP_FIELD;
 			}else{
-- 
2.19.0

